tapeagents/steps.py

```diff
@@ -0,0 +1,92 @@
+import base64
+import json
+import os
+from typing import Literal
+
+from pydantic import Field
+
+from tapeagents.core import Action, Observation, Thought
+
+################# Actions #################
+
+
+class GetVideoAction(Action):
+    """
+    Action that loads the video from the provided url and returns the subtitle and its contact sheet.
+    """
+
+    kind: Literal["get_video_action"] = "get_video_action"
+    video_url: str = Field(description="url of the video to watch")
+    start_time: str = Field(
+        description="time of the video to start watching from, if applicable, otherwise empty str", default=""
+    )
+
+
+################### Observations ###################
+
+
+class ImageObservation(Observation):
+    kind: Literal["image_observation"] = "image_observation"
+    image_path: str
+    thumbnail_path: str
+    image_caption: str
+    error: int | None = None
+
+    def llm_view_trimmed(self, indent: int | None = 2) -> list[dict]:
+        step_data = {"kind": self.kind, "image_caption": self.image_caption, "error": self.error}
+        return json.dumps(step_data, indent=indent, ensure_ascii=False)
+
+    def llm_view(self, indent: int | None = 2) -> list[dict]:
```

Comment:
llm_view() should return `str` as we past it directly to the message content. Let call this one `to_messages()` maybe?

Reply:
Aha, I see it in the OAI docs https://platform.openai.com/docs/guides/vision. So it's acceptable to pass a list of dicts into the message. We need to update our main llm_view() return signature as well

Reply:
LiteLLM uses str indeed: https://github.com/BerriAI/litellm/blob/main/litellm/types/utils.py#L394

But strangely enough, we need an array for images: https://docs.litellm.ai/docs/completion/vision, https://platform.openai.com/docs/guides/vision